app/src/main/java/io/gnosis/safe/ui/safe/settings/safe/AdvancedSafeSettingsFragment.kt

```diff
@@ -0,0 +1,139 @@
+                else -> setUi(state.isLoading)
+            }
+        })
+        binding.refresh.setOnRefreshListener {
+            viewModel.load()
+        }
+        viewModel.load()
+    }
+
+    private fun setUi(isLoading: Boolean, safeInfo: SafeInfo? = null) {
+        updateLoading(isLoading)
+        safeInfo?.let { setSafeInfo(it) }
+    }
+
+    private fun setSafeInfo(safeInfo: SafeInfo) {
+        with(binding) {
+            fallbackHandlerContainer.removeAllViews()
+            fallbackHandlerContainer.addView(fallbackHandlerView(safeInfo.fallbackHandler))
+            nonce.name = safeInfo.nonce.toString()
+            modulesContainer.removeAllViews()
+            safeInfo.modules.takeUnless { it.isEmpty() }?.let {
+                moduleLabel.visible(true)
+                it.forEach { module ->
+                    modulesContainer.addView(labeledAddress(module))
+                }
+            } ?: run { moduleLabel.visible(false) }
+        }
+    }
+
+    private fun fallbackHandlerView(fallbackHandler: Solidity.Address?): View =
+        when {
+            fallbackHandler == null || fallbackHandler.value == BigInteger.ZERO -> SettingItem(requireContext()).apply {
+                background = ContextCompat.getDrawable(requireContext(), R.drawable.background_selectable_white)
+                layoutParams = FrameLayout.LayoutParams(FrameLayout.LayoutParams.MATCH_PARENT, dpToPx(60))
+                openable = false
+                name = getString(R.string.safe_settings_not_set)
+            }
+            viewModel.isDefaultFallbackHandler(fallbackHandler) -> labeledAddress(fallbackHandler, R.string.safe_settings_default_fallback_handler)
+            else -> labeledAddress(fallbackHandler, R.string.safe_settings_unknown)
+        }
+
+    private fun labeledAddress(address: Solidity.Address, @StringRes labelId: Int): LabeledAddressItem =
+        labeledAddress(address, getString(labelId))
+
+    private fun labeledAddress(address: Solidity.Address, label: String = ""): LabeledAddressItem {
+        return LabeledAddressItem(requireContext()).apply {
+            background = ContextCompat.getDrawable(requireContext(), R.drawable.background_selectable_white)
+            layoutParams = LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT)
+            this.address = address
+            this.label = label
+        }
+    }
+
+    private fun handleError(throwable: Throwable) {
+        with(binding) {
+            refresh.isRefreshing = false
+            mainContainer.visible(false)
+            progress.visible(false)
+        }
+        snackbar(requireView(), throwable.message ?: getString(R.string.error_invalid_safe))
+        Timber.e(throwable)
+    }
+
+    private fun updateLoading(showLoading: Boolean) {
+        with(binding) {
+            refresh.isRefreshing = false
```

Comment:
not a blocker: when we have content and we do pull to refresh, showing progress bar is not necessary. Pull to refresh should be enough